### DISCUSSION TITLE
AWS IAM authentication with Quarkus-redis-client

### DISCUSSION THREAD

**Discussion by lm-gunjan** [quarkusio/quarkus#42454] [Q&A] [answered]
I am looking to use AWS IAM authentication with Quarkus-redis-client.

I came across this [issue](https://github.com/quarkusio/quarkus/issues/16284) 
it allows to set host URI programatically but doesn't seems to handle the case when password is being rotated.

 In aws [example](https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/auth-iam.html) there are credentialProviders for redis library that facilities it, does Quarkus-redis-client has anything similar.

[a comment by an automated account is not shown]

**Comment by Ladicek** [comment, marked as answer]
To be honest, I don't think the underlying Vert.x Redis client supports rotating the password. There was a PR for that, but it got stalled.

**Reply by lm-gunjan** [reply]
Any alternative reactive library that I can consider ?

**Reply by geniusit** [reply]
+1 
With lettuce it's possible, it would be super nice that vertx also provides it... there is an open [pr](https://github.com/vert-x3/vertx-redis-client/issues/356)
